### PR TITLE
#64088 | Fix Duplication of Organizers and Venues on Event Preview

### PR DESCRIPTION
**Pull request by ggwicz**
_Ref:_ [C#64088](http://central.tri.be/issues/64088)

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Whitespace found at end of line | [Main.php:3255](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3255)
:x: | Whitespace found at end of line | [Main.php:3259](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3259)
:x: | Whitespace found at end of line | [Main.php:3262](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3262)
:x: | Whitespace found at end of line | [Main.php:3264](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3264)
:x: | Whitespace found at end of line | [Main.php:3268](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3268)
:x: | Whitespace found at end of line | [Main.php:3286](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3286)
:x: | Whitespace found at end of line | [Main.php:3294](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3294)
:x: | Whitespace found at end of line | [Main.php:3298](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3298)
:x: | Whitespace found at end of line | [Main.php:3310](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3310)
:x: | Whitespace found at end of line | [Main.php:3312](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3312)
:x: | Whitespace found at end of line | [Main.php:3316](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3316)
:x: | Whitespace found at end of line | [Main.php:3318](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3318)
:x: | Whitespace found at end of line | [Main.php:3320](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3320)
:x: | Whitespace found at end of line | [Main.php:3322](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3322)
:x: | Whitespace found at end of line | [Main.php:3327](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3327)
:x: | Whitespace found at end of line | [Main.php:3340](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3340)
:x: | Whitespace found at end of line | [Main.php:3342](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3342)
:x: | Whitespace found at end of line | [Main.php:3344](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3344)
:x: | Whitespace found at end of line | [Main.php:3348](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3348)
:x: | Whitespace found at end of line | [Main.php:3361](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3361)
:x: | Whitespace found at end of line | [Main.php:3363](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3363)
:x: | Whitespace found at end of line | [Main.php:3367](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3367)
:x: | Whitespace found at end of line | [Main.php:3371](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3371)
:x: | Whitespace found at end of line | [Main.php:3384](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3384)
:x: | Whitespace found at end of line | [Main.php:3386](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3386)
:x: | Whitespace found at end of line | [Main.php:3390](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3390)
:x: | Whitespace found at end of line | [Main.php:3392](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3392)
:x: | Whitespace found at end of line | [Main.php:3396](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3396)
:x: | Whitespace found at end of line | [Main.php:3400](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3400)
:x: | Whitespace found at end of line | [Main.php:3417](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3417)
:x: | Whitespace found at end of line | [Main.php:3421](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3421)
:x: | Whitespace found at end of line | [Main.php:3423](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3423)
:x: | Whitespace found at end of line | [Main.php:3427](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3427)
:x: | Whitespace found at end of line | [Main.php:3431](https://github.com/moderntribe/the-events-calendar/blob/6dc122b22631b8b6e7de2f17edeb81f0c3adb975/src/Tribe/Main.php#L3431)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Main.php [hunk 1]
```diff
@@ -3054,6 +3057,10 @@ public function publishAssociatedTypes( $post_id, $post ) {
 			remove_action( 'save_post_' . self::VENUE_POST_TYPE, array( $this, 'save_venue_data' ), 16, 2 );
 			remove_action( 'save_post_' . self::ORGANIZER_POST_TYPE, array( $this, 'save_organizer_data' ), 16, 2 );
 
+			// Remove any "preview" venues and organizers (duplicates) attached to this event.
+			$this->remove_preview_venues( $post_id, true );
```
Comment (bordoni) [inline src/Tribe/Main.php]: Can we attach these to an action, so it's not running on `plugins_loaded` where we don't have a WP User set yet? Maybe `admin_init` or `init`

Comment (ggwicz) [inline src/Tribe/Main.php]: can you clarify what you mean @gustavo? the hook is "hooked" to the publish action in a method that is itself running `plugins_loaded`, but this `remove_preview_venues` function itself doesn't run on `plugins_loaded` -- right? apologies if i'm missing something here -- this function runs on the `publish_{post_type}` action *only*

Comment (bordoni) [inline src/Tribe/Main.php]: Sorry, I hadn't expanded the code when I initially looked, this looks good for a merge.

**Pull request merged**